### PR TITLE
Version: 2021.04.00

### PR DESCRIPTION
**Pull request by ax3l**
Add the latest release.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ax3l** [comment]
- [x] re-render after https://github.com/conda-forge/blaspp-feedstock/pull/5 was merged & uploaded

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,7 +27,8 @@ requirements:
     - ninja          # [win]
     - cmake >=3.15.0
   host:
-    - blaspp =={{ version }}
+    #- blaspp =={{ version }}
+    - blaspp ==2021.04.01
```
Comment (ax3l) [inline recipe/meta.yaml]:
```suggestion
    - blaspp ==2021.04
```

**Comment by ax3l** [comment]
- [x] wait for https://github.com/conda-forge/blaspp-feedstock/pull/4 to be merged, as this changes as least on OSX x64 the picked up BLAS libs
- [ ] potentially enable macOS aarch64 as well

File: recipe/build.sh [hunk 2]
```diff
@@ -7,6 +7,7 @@ cmake -S . -B build             \
     -DCMAKE_VERBOSE_MAKEFILE=ON \
     -DBUILD_SHARED_LIBS=ON      \
     -Dbuild_tests=OFF           \
+    -DLAPACK_LIBRARIES=${PREFIX}/lib/liblapack${SHLIB_EXT};${PREFIX}/lib/libblas${SHLIB_EXT} \
```
Comment (ax3l) [inline recipe/build.sh]:
```suggestion
    -DLAPACK_LIBRARIES=-llapack \
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -35,7 +27,9 @@ requirements:
     - ninja          # [win]
     - cmake >=3.15.0
   host:
-    - blaspp =={{ version }}
+    #- blaspp =={{ version }}
+    - blaspp ==2021.04
+    - libblas
     - liblapack
     - liblapacke
```
Comment (ax3l) [inline recipe/meta.yaml:34]:
```suggestion
```

File: recipe/build.sh [hunk 4]
```diff
@@ -7,6 +7,7 @@ cmake -S . -B build             \
     -DCMAKE_VERBOSE_MAKEFILE=ON \
     -DBUILD_SHARED_LIBS=ON      \
     -Dbuild_tests=OFF           \
+    -DLAPACK_LIBRARIES=-llapack \
```
Comment (ax3l) [inline recipe/build.sh]:
```suggestion
```

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/build.sh [hunk 5]
```diff
@@ -7,7 +7,7 @@ cmake -S . -B build             \
     -DCMAKE_VERBOSE_MAKEFILE=ON \
     -DBUILD_SHARED_LIBS=ON      \
     -Dbuild_tests=OFF           \
-    ${CMAKE_ARGS}
+    ${CMAKE_ARGS} || (cat CMakeFiles/CMakeOutput.log && CMakeFiles/CMakeError.log && false)
```
Comment (h-vetinari) [inline recipe/build.sh]: Probably missing a 🐱 for `CMakeError.log`

```suggestion
    ${CMAKE_ARGS} || (cat CMakeFiles/CMakeOutput.log && cat CMakeFiles/CMakeError.log && false)
```

File: recipe/build.sh [hunk 6]
```diff
@@ -7,7 +7,7 @@ cmake -S . -B build             \
     -DCMAKE_VERBOSE_MAKEFILE=ON \
     -DBUILD_SHARED_LIBS=ON      \
     -Dbuild_tests=OFF           \
-    ${CMAKE_ARGS}
+    ${CMAKE_ARGS} || (cat CMakeFiles/CMakeOutput.log && cat CMakeFiles/CMakeError.log && false)
```
Comment (ax3l) [inline recipe/build.sh]:
```suggestion
    ${CMAKE_ARGS} || { cat build/CMakeFiles/CMakeOutput.log; cat build/CMakeFiles/CMakeError.log; exit 1; }
```

File: recipe/meta.yaml [hunk 7]
```diff
@@ -35,9 +27,10 @@ requirements:
     - ninja          # [win]
     - cmake >=3.15.0
   host:
-    - blaspp =={{ version }}
+    #- blaspp =={{ version }}
+    - blaspp ==2021.04
+    - libblas
```
Comment (ax3l) [inline recipe/meta.yaml]: remove again: `libblas`
```suggestion
```

not really needed here, was just trying around

**Comment by ax3l** [comment]
Hm, I don't see it yet...

**Comment by ax3l** [comment]
Logic seems to reside here:
- https://bitbucket.org/icl/lapackpp/src/2021.04.00/cmake/LAPACKFinder.cmake#lines-113:142
- https://bitbucket.org/icl/lapackpp/src/2021.04.00/config/lapack_pstrf.cc
- https://bitbucket.org/icl/lapackpp/src/2021.04.00/config/config.h

**Comment by ax3l** [comment]
I might have an idea: https://bitbucket.org/icl/lapackpp/issues/12/

**Comment by isuruf** [comment]
@conda-forge-admin, rerender.

File: recipe/build.sh [hunk 8]
```diff
@@ -1,13 +1,16 @@
 #!/bin/bash
 
+# work-around https://bitbucket.org/icl/lapackpp/issues/12
+export CXXFLAGS="${CXXFLAGS} -DFORTRAN_ADD_"
```
Comment (ax3l) [inline recipe/build.sh]:
```suggestion
export CXXFLAGS="${CXXFLAGS} -DLAPACK_FORTRAN_ADD_"
```

File: recipe/bld.bat [hunk 9]
```diff
@@ -1,5 +1,8 @@
 @echo on
 
+:: work-around https://bitbucket.org/icl/lapackpp/issues/12
+set CXXFLAGS=%CXXFLAGS% -DFORTRAN_ADD_
```
Comment (ax3l) [inline recipe/bld.bat]:
```suggestion
set CXXFLAGS=%CXXFLAGS% -DLAPACK_FORTRAN_ADD_
```

**Comment by ax3l** [comment]
@teonnik @Sely85 @G-Ragghianti @mgates3
LAPACK++ 2021.04.00 has some serious troubles with the heuristics to find LAPACK. Is it possible that it is expected to run that Python config script in your repo before installing it? I cannot spot such as step in Spack at least. We have to work-around by setting the fortran mangling externally due to https://bitbucket.org/icl/lapackpp/issues/12

Maybe you know more, can you help? :) The trouble with the find LAPACK logic started when updating to 2021.04.00.

Some background how conda-forge handles LAPACK variants: https://github.com/conda-forge/lapack-feedstock/issues/57#issuecomment-989335501

File: recipe/meta.yaml [hunk 10]
```diff
@@ -35,7 +27,10 @@ requirements:
...
+    - libcblas
```
Comment (ax3l) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 11]
```diff
@@ -35,7 +27,8 @@ requirements:
     - ninja          # [win]
     - cmake >=3.15.0
   host:
-    - blaspp =={{ version }}
+    #- blaspp =={{ version }}
+    - blaspp ==2021.04
     - liblapack
```
Comment (ax3l) [inline recipe/meta.yaml:33]: Weird:
```suggestion
    - libblas  # [win]
    - liblapack
```
```
(%BUILD_PREFIX%) %SRC_DIR%>cmake --build build --config Release --parallel 2 
ninja: error: 'D:/bld/lapackpp_1639114017583/_h_env/Library/lib/blas.lib', needed by 'lapackpp.dll', missing and no known rule to make it
```

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender.

**Comment by ax3l** [comment]
@h-vetinari is LAPACK already migrated to osx_arm64? :)

**Comment by h-vetinari** [comment]
> @h-vetinari is LAPACK already migrated to osx_arm64? :)

Yes, otherwise it'd be difficult to build anything from scipy on up (the only flavours are netlib and openblas)... 🙃

I'm on a mobile right now so cannot look in more detail at the CI.

**Comment by ax3l** [comment]
Ah I see the problem, since LAPACK++ currently uses `try_run` for checks of dependencies, it cannot yet be cross-compiled:
```
   -llapack                                         no (didn't run: int mismatch, etc.)
   LAPACK library not found.
```

Can be generalized upstream cc @mgates3 :)

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender.

**Pull request merged**